sdk/src/main/java/com/hedera/hashgraph/sdk/NodeCreateTransaction.java

```diff
@@ -0,0 +1,439 @@
+
+    /**
+     * Extract the description of the node.
+     * @return the node's description.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Sets the description of the node.
+     * @param description The String to be set as the description of the node.
+     * @return {@code this}
+     */
+    public NodeCreateTransaction setDescription(String description) {
+        requireNotFrozen();
+        Objects.requireNonNull(description);
+        this.description = description;
+        return this;
+    }
+
+    /**
+     * Extract the list of service endpoints for gossip.
+     * @return the list of service endpoints for gossip.
+     */
+    public List<Endpoint> getGossipEndpoints() {
+        return gossipEndpoints;
+    }
+
+    /**
+     * Assign the list of service endpoints for gossip.
+     * @param gossipEndpoints the list of service endpoints for gossip.
+     * @return {@code this}
+     */
+    public NodeCreateTransaction setGossipEndpoints(List<Endpoint> gossipEndpoints) {
+        requireNotFrozen();
+        Objects.requireNonNull(gossipEndpoints);
+        this.gossipEndpoints = new ArrayList<>(gossipEndpoints);
+        return this;
+    }
+
+    /**
+     * Add an endpoint for gossip to the list of service endpoints for gossip.
+     * @param gossipEndpoint endpoints for gossip to add.
+     * @return {@code this}
+     */
+    public NodeCreateTransaction addGossipEndpoint(Endpoint gossipEndpoint) {
+        requireNotFrozen();
+        gossipEndpoints.add(gossipEndpoint);
+        return this;
+    }
+
+    /**
+     * Extract the list of service endpoints for gRPC calls.
+     * @return the list of service endpoints for gRPC calls.
+     */
+    public List<Endpoint> getServiceEndpoints() {
+        return serviceEndpoints;
+    }
+
+    /**
+     * Assign the list of service endpoints for gRPC calls.
+     * @param serviceEndpoints list of service endpoints for gRPC calls.
+     * @return {@code this}
+     */
+    public NodeCreateTransaction setServiceEndpoints(List<Endpoint> serviceEndpoints) {
+        requireNotFrozen();
+        Objects.requireNonNull(serviceEndpoints);
+        this.serviceEndpoints = new ArrayList<>(serviceEndpoints);
+        return this;
+    }
+
+    /**
+     * Add an endpoint for gRPC calls to the list of service endpoints for gRPC calls.
+     * @param serviceEndpoint endpoints for gRPC calls to add.
+     * @return {@code this}
+     */
+    public NodeCreateTransaction addServiceEndpoint(Endpoint serviceEndpoint) {
+        requireNotFrozen();
+        serviceEndpoints.add(serviceEndpoint);
+        return this;
+    }
+
+    /**
+     * Extract the certificate used to sign gossip events.
+     * @return the DER encoding of the certificate presented.
+     */
+    @Nullable
+    public byte[] getGossipCaCertificate() {
+        return gossipCaCertificate != null ? Arrays.copyOf(gossipCaCertificate, gossipCaCertificate.length) : null;
```

Comment:
What is this logic for?

Reply:
Possibility for `gossipCaCertificate` to be `null`. And it is needed when we are building the transaction body (we won't set this value if it is `null`). See similar logic in `ContractCreateTransaction`: https://github.com/hashgraph/hedera-sdk-java/blob/546b49b8f95387666a99800e344153c3c3dddf8f/sdk/src/main/java/com/hedera/hashgraph/sdk/ContractCreateTransaction.java#L183